### PR TITLE
NO-JIRA: Use oc rhel8 binary explicitly until base image is rhel9

### PR DESCRIPTION
**Pull request by ardaguclu**
must-gather includes oc invocations to gather some data and must-gather gets this oc from cli image. However, when we try to bump cli image to RHEL9, since must-gather is still on RHEL8, all oc invocations failed. This PR changes to use cli-artifacts image instead of cli. Thanks to that change, must-gather can explicitly uses oc.rhel8 and that decouples the cli and must-gather images for the future similar migrations.

example failures;

*  https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1752/pull-ci-openshift-oc-master-e2e-aws-ovn/1784837264291926016
* https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1752/pull-ci-openshift-oc-master-e2e-agnostic-ovn-cmd/1784837260940677120

### CODE REVIEW

**Comment by sferich888** [comment]
While I don't like doing this; this is temporary (to work around SSL compatibility issue with FIPS across OCP versions). 

/lgtm

**Review by sferich888 — APPROVED** [review]

**Comment by ardaguclu** [comment]
/hold
until dependant merges

**Pull request merged**
